finality-aleph/src/nodes.rs

```diff
@@ -113,17 +120,36 @@ where
         map_updater.run().await
     });
 
-    let (authority_justification_tx, handler_task) =
-        setup_justification_handler(JustificationParams {
-            justification_rx,
-            network,
-            client: client.clone(),
-            blockchain_backend,
-            metrics: metrics.clone(),
-            session_period,
-            millisecs_per_block,
-            session_map: session_authorities.clone(),
-        });
+    let chain_events = SubstrateChainStatusNotifier::new(
+        client.finality_notification_stream(),
+        client.import_notification_stream(),
+    );
+
+    let genesis_header = match chain_status.finalized_at(0) {
+        Ok(Some(justification)) => justification.header().clone(),
+        _ => panic!("the genesis block should be finalized"),
+    };
+    let verifier = VerifierCache::new(
+        session_period,
+        SubstrateFinalizationInfo::new(client.clone()),
+        AuthorityProviderImpl::new(client.clone()),
+        VERIFIER_CACHE_SIZE,
+        genesis_header,
+    );
+    let finalizer = AlephFinalizer::new(client.clone());
+    let (sync_service, justifications_for_sync) = match SyncService::new(
+        block_sync_network,
+        chain_events,
+        chain_status.clone(),
+        verifier,
+        finalizer,
+        session_period,
+        justification_rx,
+    ) {
+        Ok(x) => x,
+        Err(e) => panic!("Failed to run Sync service: {}", e),
```

Comment:
nit: `Failed to initialize...`? Run is called later.

Reply:
Good point, fixed.